### PR TITLE
node-515-survey-administration-crud-operations

### PR DESCRIPTION
**Pull request by chadiegilnerubia**
Ticket ID:
Create API to Create / Retrieve / Update / Delete Survey Administration
[#515](https://github.com/nerubia/kh360-nodejs/issues/515)

### CODE REVIEW

File: src/controllers/admin/survey-administration-controller.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+export const destroy = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params
+    const survey = await SurveyAdministrationService.getById(parseInt(id))
+    if (survey === null) {
+      return res.status(400).json({ message: "Invalid id" })
+    }
+    if (survey.status !== SurveyAdministrationStatus.Draft) {
+      return res.status(403).json({ message: "This action is not allowed" })
+    }
```
Comment (emmaacha) [inline src/controllers/admin/survey-administration-controller.ts:170]: Please put these inside survey-administration-service

File: src/controllers/admin/survey-administration-controller.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    } = req.body
+    await createSurveyAdministrationSchema.validate({
+      name,
+      survey_start_date,
+      survey_end_date,
+      survey_template_id: parseInt(survey_template_id),
+      remarks,
+      email_subject,
+      email_content,
+      status,
```
Comment (emmaacha) [inline src/controllers/admin/survey-administration-controller.ts:128]: I think status should not be manually updated here in this api, we can remove status here

File: src/controllers/admin/survey-administration-controller.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
```
Comment (emmaacha) [inline src/controllers/admin/survey-administration-controller.ts:71]: We won't be passing status here so we can remove this

File: src/controllers/admin/survey-administration-controller.ts [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+    const updateSurvey = await SurveyAdministrationService.updateById(parseInt(id), {
+      name: name as string,
+      survey_start_date: survey_start_date as Date,
+      survey_end_date: survey_end_date as Date,
+      survey_template_id: parseInt(survey_template_id),
+      remarks: remarks as string,
+      email_subject: email_subject as string,
+      email_content: email_content as string,
+      status: status as string,
```
Comment (emmaacha) [inline src/controllers/admin/survey-administration-controller.ts:143]: We can remove this

**Review by emmaacha — CHANGES_REQUESTED** [review]

File: src/routes/admin/survey-administration.ts
Comment (emmaacha) [inline src/routes/admin/survey-administration.ts:1]: Kindly rename this to `survey-administration-route.ts`

**Review by emmaacha — CHANGES_REQUESTED** [review]

File: src/controllers/admin/survey-administration-controller.ts [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      survey_template_id: parseInt(survey_template_id),
+      remarks,
+      email_subject,
+      email_content,
+    })
+    const survey = await SurveyAdministrationService.getById(parseInt(id))
+    if (survey === null) {
+      return res.status(400).json({ message: "Invalid id." })
+    }
+
```
Comment (emmaacha) [inline src/controllers/admin/survey-administration-controller.ts:128]: Please put this in survey administration service

**Review by emmaacha — CHANGES_REQUESTED** [review]

**Review by emmaacha — APPROVED** [review]

File: src/controllers/admin/survey-administration-controller.ts [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    const newSurvey = await SurveyAdministrationService.create({
+      name: name as string,
+      survey_start_date: new Date(survey_start_date),
+      survey_end_date: new Date(survey_end_date),
+      survey_template_id: parseInt(survey_template_id),
+      remarks,
+      email_subject,
+      email_content,
+    })
```
Comment (emmaacha) [inline src/controllers/admin/survey-administration-controller.ts:79]: Kindly add status: SurveyAdministrationStatus.Draft here to auto set the status to Draft

**Review by emmaacha — CHANGES_REQUESTED** [review]

**Review by emmaacha — APPROVED** [review]

**Review by gborgonia-ir — APPROVED** [review]

**Pull request merged**